docs/authentication/ec2-metadata.md

```diff
@@ -10,5 +10,5 @@ When you run code within an EC2 instance (or EKS, Lambda), AsyncAws is able to f
 When running a single application on the Server, this is the simplest way to grant permissions to the application. You
 have nothing to configure on the application, you only grant permissions on the Role attached to the instance.
 
-AsyncAWS uses the IMDSv1 protocol to read instance metadata. So this authentication method won't work if the v1
+AsyncAWS uses the IMDSv2 protocol to read instance metadata. So this authentication method won't work if the v2
```

Comment:
That's a BC break for people using AsyncAws with `IMDSv1` and disabling `IMDSv2`
Is there a way to try `IMDSv2` and deal with errors (from the call to `TOKEN_ENDPOINT`) to decide if we could fallback to `IMDSv1~

Reply:
I adjusted the code. This is also how the AWS documentation describes how their SDKs work:

> The AWS SDKs use IMDSv2 calls by default. If the IMDSv2 call receives no response, the SDK retries the call and, if still unsuccessful, uses IMDSv1. This can result in a delay.

I only left out the retry because the lib uses a retry http client decorator.